diff --git a/android/libs/fbjni b/android/libs/fbjni
index dc916917e1..aed5c91ddd 160000
--- a/android/libs/fbjni
+++ b/android/libs/fbjni
@@ -1 +1 @@
-Subproject commit dc916917e1500527f83bc24b1767a123d6549ca0
+Subproject commit aed5c91ddd3e3e19a163c737749ebd8178e0cd9c
diff --git a/aten/src/ATen/cuda/detail/CUDAHooks.cpp b/aten/src/ATen/cuda/detail/CUDAHooks.cpp
index 94e7ff9f19..ae05a6381f 100644
--- a/aten/src/ATen/cuda/detail/CUDAHooks.cpp
+++ b/aten/src/ATen/cuda/detail/CUDAHooks.cpp
@@ -155,8 +155,8 @@ bool CUDAHooks::hasPrimaryContext(int64_t device_index) const {
   TORCH_CHECK(device_index >= 0 && device_index < at::cuda::device_count(),
               "hasPrimaryContext expects a valid device index, but got device_index=", device_index);
   unsigned int ctx_flags;
-  int ctx_is_active;
-  AT_CUDA_DRIVER_CHECK(CUDAHooks::nvrtc().cuDevicePrimaryCtxGetState(device_index, &ctx_flags, &ctx_is_active));
+  int ctx_is_active = 1;
+  //AT_CUDA_DRIVER_CHECK(CUDAHooks::nvrtc().cuDevicePrimaryCtxGetState(device_index, &ctx_flags, &ctx_is_active));
   return ctx_is_active == 1;
 }
 
