diff --git a/CMakeLists.txt b/CMakeLists.txt
index de00f6a..a4d6fd8 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -27,3 +27,20 @@ target_include_directories(
   PRIVATE
   ${CMAKE_CURRENT_SOURCE_DIR}
   )
+
+install(TARGETS encode decode
+  DESTINATION
+    bin
+  )
+install(FILES coder.h io_torch.h
+  DESTINATION
+    include/obfuscator
+  )
+install(FILES encode_rodata.py mark_rodata.lds minimal_pybind.map
+  DESTINATION
+    lib/obfuscator
+  )
+install(FILES obfuscator.cmake
+  DESTINATION
+    share/cmake/obfuscator
+  )
diff --git a/obfuscator.cmake b/obfuscator.cmake
index 9ef237b..62bad30 100644
--- a/obfuscator.cmake
+++ b/obfuscator.cmake
@@ -3,9 +3,8 @@ function(embed_encoded_object name seed)
     OUTPUT
     ${name}.enc.o
     COMMAND
-    ${CMAKE_CURRENT_BINARY_DIR}/obfuscator/encode ${seed} ${name} ${name}.enc && objcopy -I binary -O elf64-x86-64 -B i386:x86-64 ${name}.enc ${name}.enc.o && rm ${name}.enc
+    ${OBFUSCATOR_INSTALL_PREFIX}/bin/encode ${seed} ${name} ${name}.enc && objcopy -I binary -O elf64-x86-64 -B i386:x86-64 ${name}.enc ${name}.enc.o && rm ${name}.enc
     MAIN_DEPENDENCY ${name}
-    DEPENDS encode
     )
 endfunction()
 
@@ -16,8 +15,8 @@ function(encode_rodata out_name in_name seed)
     OUTPUT
     ${out_name}
     COMMAND
-    cp ${in_so} ${out_so} && ${PYTHON_EXECUTABLE} ${CMAKE_CURRENT_SOURCE_DIR}/obfuscator/encode_rodata.py --seed ${seed} --binary ${out_so} --encode_bin ${CMAKE_CURRENT_BINARY_DIR}/obfuscator/encode
-    DEPENDS ${in_name} ${CMAKE_CURRENT_SOURCE_DIR}/obfuscator/encode_rodata.py ${CMAKE_CURRENT_BINARY_DIR}/obfuscator/encode
+    cp ${in_so} ${out_so} && ${PYTHON_EXECUTABLE} ${OBFUSCATOR_INSTALL_PREFIX}/lib/obfuscator/encode_rodata.py --seed ${seed} --binary ${out_so} --encode_bin ${OBFUSCATOR_INSTALL_PREFIX}/bin/encode
+    DEPENDS ${in_name}
     )
   add_custom_target(
     build_${out_name}
@@ -25,3 +24,5 @@ function(encode_rodata out_name in_name seed)
     DEPENDS ${out_name}
     )
 endfunction()
+
+set(OBFUSCATOR_INCLUDE_DIRS ${OBFUSCATOR_INSTALL_PREFIX}/include)
