diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index 580fd68..30ff536 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -2,46 +2,48 @@ add_subdirectory(internal)
 
 add_library(${PROJECT_NAME}_loop_generic STATIC loop.cc)
 target_compile_definitions(${PROJECT_NAME}_loop_generic PRIVATE PFNMS_TARGET_GENERIC)
-set(${PROJECT_NAME}_LIBS ${${PROJECT_NAME}_LIBS} ${PROJECT_NAME}_loop_generic)
+set(${PROJECT_NAME}_SRC_LIBS ${${PROJECT_NAME}_SRC_LIBS} ${PROJECT_NAME}_loop_generic)
 
 if(${PROJECT_NAME}_x86)
     add_library(${PROJECT_NAME}_loop_avx512 STATIC loop.cc)
     target_compile_options(${PROJECT_NAME}_loop_avx512 PRIVATE -mavx512f -mavx512dq -mbmi)
     target_compile_definitions(${PROJECT_NAME}_loop_avx512 PRIVATE PFNMS_TARGET_AVX512)
-    set(${PROJECT_NAME}_LIBS ${${PROJECT_NAME}_LIBS} ${PROJECT_NAME}_loop_avx512)
+    set(${PROJECT_NAME}_SRC_LIBS ${${PROJECT_NAME}_SRC_LIBS} ${PROJECT_NAME}_loop_avx512)
 endif()
 
 if(${PROJECT_NAME}_x86)
     add_library(${PROJECT_NAME}_loop_avx STATIC loop.cc)
     target_compile_options(${PROJECT_NAME}_loop_avx PRIVATE -mavx -mfma -mbmi)
     target_compile_definitions(${PROJECT_NAME}_loop_avx PRIVATE PFNMS_TARGET_AVX)
-    set(${PROJECT_NAME}_LIBS ${${PROJECT_NAME}_LIBS} ${PROJECT_NAME}_loop_avx)
+    set(${PROJECT_NAME}_SRC_LIBS ${${PROJECT_NAME}_SRC_LIBS} ${PROJECT_NAME}_loop_avx)
 endif()
 
 if(${PROJECT_NAME}_aarch64)
     add_library(${PROJECT_NAME}_loop_neon STATIC loop.cc)
     target_compile_definitions(${PROJECT_NAME}_loop_neon PRIVATE PFNMS_TARGET_NEON)
-    set(${PROJECT_NAME}_LIBS ${${PROJECT_NAME}_LIBS} ${PROJECT_NAME}_loop_neon)
+    set(${PROJECT_NAME}_SRC_LIBS ${${PROJECT_NAME}_SRC_LIBS} ${PROJECT_NAME}_loop_neon)
 endif()
 
 if(${PROJECT_NAME}_arm)
     add_library(${PROJECT_NAME}_loop_neon STATIC loop.cc)
     target_compile_options(${PROJECT_NAME}_loop_neon PRIVATE -mfpu=neon)
     target_compile_definitions(${PROJECT_NAME}_loop_neon PRIVATE PFNMS_TARGET_NEON)
-    set(${PROJECT_NAME}_LIBS ${${PROJECT_NAME}_LIBS} ${PROJECT_NAME}_loop_neon)
+    set(${PROJECT_NAME}_SRC_LIBS ${${PROJECT_NAME}_SRC_LIBS} ${PROJECT_NAME}_loop_neon)
 endif()
 
 add_library(${PROJECT_NAME}_sort_generic STATIC sort.cc)
 target_compile_definitions(${PROJECT_NAME}_sort_generic PRIVATE PFNMS_TARGET_GENERIC)
-set(${PROJECT_NAME}_LIBS ${${PROJECT_NAME}_LIBS} ${PROJECT_NAME}_sort_generic)
+set(${PROJECT_NAME}_SRC_LIBS ${${PROJECT_NAME}_SRC_LIBS} ${PROJECT_NAME}_sort_generic)
 
 if(${PROJECT_NAME}_x86)
     add_library(${PROJECT_NAME}_sort_bmi1 STATIC sort.cc)
     target_compile_options(${PROJECT_NAME}_sort_bmi1 PRIVATE -mbmi)
     target_compile_definitions(${PROJECT_NAME}_sort_bmi1 PRIVATE PFNMS_TARGET_BMI1)
-    set(${PROJECT_NAME}_LIBS ${${PROJECT_NAME}_LIBS} ${PROJECT_NAME}_sort_bmi1)
+    set(${PROJECT_NAME}_SRC_LIBS ${${PROJECT_NAME}_SRC_LIBS} ${PROJECT_NAME}_sort_bmi1)
 endif()
 
+set(${PROJECT_NAME}_LIBS ${${PROJECT_NAME}_LIBS} ${${PROJECT_NAME}_SRC_LIBS})
+
 add_library(pfnms pfnms.cc)
 target_include_directories(pfnms PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/../include)
 target_link_libraries(pfnms PRIVATE ${${PROJECT_NAME}_LIBS})
@@ -68,3 +70,6 @@ if(${${PROJECT_NAME}_ENABLE_OPENMP})
     target_link_libraries(pfnms PRIVATE ${OpenMP_CXX_LIB_NAMES})
     target_compile_definitions(pfnms PRIVATE -DPFNMS_ENABLE_OPENMP=1)
 endif()
+
+install(TARGETS pfnms ${${PROJECT_NAME}_SRC_LIBS}
+  ARCHIVE DESTINATION lib)
diff --git a/src/internal/CMakeLists.txt b/src/internal/CMakeLists.txt
index c9c7aa2..618fcb6 100644
--- a/src/internal/CMakeLists.txt
+++ b/src/internal/CMakeLists.txt
@@ -18,3 +18,5 @@ if(${${PROJECT_NAME}_build_runtime})
 endif()
 
 set(${PROJECT_NAME}_LIBS ${${PROJECT_NAME}_LIBS} PARENT_SCOPE)
+install(TARGETS ${PROJECT_NAME}_runtime
+  ARCHIVE DESTINATION lib)
