diff --git a/cmake/pfvm-config.cmake b/cmake/pfvm-config.cmake
index 7e143bc..78c05a0 100644
--- a/cmake/pfvm-config.cmake
+++ b/cmake/pfvm-config.cmake
@@ -7,22 +7,22 @@ else()
   get_filename_component(PFVM_INSTALL_PREFIX "${CMAKE_CURRENT_LIST_DIR}/../../../" ABSOLUTE)
 endif()
 
-foreach(l pfvm_runtime pfvm_compiler pfvm_common pfvm_tools)
-  find_library(found_lib ${l}
+foreach(l pfvm_runtime pfvm_compiler pfvm_configs pfvm_common pfvm_tools)
+  find_library(found_lib_${l} ${l}
     PATHS ${PFVM_INSTALL_PREFIX}
     PATH_SUFFIXES /lib
     )
-  if(found_lib)
-    list(APPEND PFVM_LIBRARIES ${found_lib})
+  if(found_lib_${l})
+    list(APPEND PFVM_LIBRARIES ${found_lib_${l}})
   endif()
 endforeach()
 
 foreach(l pfvm_runtime_static)
-  find_library(found_lib ${l}
+  find_library(found_lib_${l} ${l}
     PATHS ${PFVM_INSTALL_PREFIX}
     PATH_SUFFIXES /lib
     )
-  list(APPEND PFVM_RUNTIME_LIBRARIES ${found_lib})
+  list(APPEND PFVM_RUNTIME_LIBRARIES ${found_lib_${l}})
 endforeach()
 
 find_path(PFVM_INCLUDE_DIRS NAMES pfvm/runtime/serializer.h
diff --git a/compiler/CMakeLists.txt b/compiler/CMakeLists.txt
index 458da56..223b111 100644
--- a/compiler/CMakeLists.txt
+++ b/compiler/CMakeLists.txt
@@ -297,4 +297,10 @@ install(
   DESTINATION include/pfvm/compiler
   COMPONENT dev
 )
+install(
+  FILES
+    atenvm/emitter.h
+  DESTINATION include/pfvm/compiler/atenvm
+  COMPONENT dev
+)
 install(TARGETS pfvm_compiler ARCHIVE DESTINATION lib)
diff --git a/compiler/gradient_ops.cc b/compiler/gradient_ops.cc
index bd8d4f4..92287de 100644
--- a/compiler/gradient_ops.cc
+++ b/compiler/gradient_ops.cc
@@ -1497,6 +1497,7 @@ bool AddGradientForNode(Graph* graph, Graph* dest_graph, Node* node, std::map<Va
         register_grad_fn(Node::kOneHot, &DoNothingGradFn);
         register_grad_fn(Node::kSequenceLength, &DoNothingGradFn);
         register_grad_fn(Node::kShape, &DoNothingGradFn);
+        register_grad_fn(Node::kNonZero, &DoNothingGradFn);
     }
 
     auto found = s_gradient_funcs->find(node->op_type());
diff --git a/configs/CMakeLists.txt b/configs/CMakeLists.txt
index d72bc8d..52d4467 100644
--- a/configs/CMakeLists.txt
+++ b/configs/CMakeLists.txt
@@ -68,3 +68,5 @@ if(PFVM_BUILD_TESTS)
     WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/..
     )
 endif()
+
+install(TARGETS pfvm_configs ARCHIVE DESTINATION lib)
